barcodes_generator_abstract/wizard/barcode_image_generate.py

```diff
@@ -0,0 +1,123 @@
+# -*- coding: utf-8 -*-
+# © 2017 Akretion (Alexis de Lattre <alexis.delattre@akretion.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+from io import BytesIO
+import logging
+logger = logging.getLogger(__name__)
+
+try:
+    import barcode
+except ImportError:
+    logger.debug('Cannot import barcode')
+
+
+class BarcodeImageGenerate(models.TransientModel):
+    _name = 'barcode.image.generate'
+    _description = 'Wizard to generate a barcode image'
+
+    @api.model
+    def _supported_barcode_formats(self):
+        res = []
+        for barcode_format in barcode.PROVIDED_BARCODES:
+            res.append((barcode_format, barcode_format))
+        return res
+
+    state = fields.Selection([
+        ('config', 'config'),
+        ('download', 'download'),
+        ], default='config')
+    image_format = fields.Selection([
+        ('PNG', 'PNG'),
+        ('JPEG', 'JPEG'),
+        ], string='Image Format', default='PNG', required=True)
+    # Idea for improvement: dynamically set the list of all supported
+    # image formats from PIL ?
+    dpi = fields.Integer(string='Image Size (DPI)', default=200, required=True)
+    quiet_zone = fields.Float(
+        string='Side Space', default=1.0,
+        help="Size of the white space on the left and on the right of the "
+             "barcode")
+    write_text = fields.Boolean(
+        string='Print Barcode Text', default=True,
+        help="Print the text of the barcode under the image")
+    text_distance = fields.Float(
+        default=2.0,
+        string='Space Between Barcode and Text',
+        help="Size of the white space between the bottom of the barcode and "
+             "the text of the barcode")
+    module_height = fields.Float(
+        default=15, string='Barcode Height',
+        help="Vertical size of the barcode")
+    module_width = fields.Float(
+        default=0.2, string='Barcode Width',
+        help="It doesn't seem to have any effect in EAN13...")
```

Comment:
Sorry, I'm rechecking this PR. I don't understand this helper. do you mean this field is useless : 
- allways ? 
- for ean13 ? 

in all cases, i don't think that this helper will help end user. Maybe a comment should be more usefull. (and maybe make this field hidden, if it is useless. don't you think ?